gbpservice/nfp/modules/configurator.py

```diff
@@ -0,0 +1,458 @@
+
+from gbpservice.nfp.configurator.lib import constants
+from gbpservice.nfp.configurator.lib import demuxer
+from oslo_log import log
+from gbpservice.nfp.core import rpc
+from gbpservice.nfp.core import fifo as queue
+from gbpservice.nfp.configurator.lib import utils
+
+AGENTS_PKG = 'gbpservice.nfp.configurator.agents'
+CONFIGURATOR_RPC_TOPIC = 'configurator'
+LOG = log.getLogger(__name__)
+
+"""Implements procedure calls invoked by an REST server.
+
+Implements following RPC methods.
+  - create_network_device_config
+  - delete_network_device_config
+  - update_network_device_config
+  - create_network_service_config
+  - delete_network_service_config
+  - update_network_service_config
+  - get_notifications
+Also implements local methods for supporting RPC methods
+
+"""
+
+
+class ConfiguratorRpcManager(object):
+
+    def __init__(self, sc, cm, conf, demuxer):
+        self.sc = sc
+        self.cm = cm
+        self.conf = conf
+        self.demuxer = demuxer
+
+    def _get_service_agent_instance(self, service_type):
+        """Provides service agent instance based on service type.
+
+        :param service_type: firewall/vpn/loadbalancer/generic_config
+
+        Returns: Instance of service agent for a given service type
+
+        """
+
+        return self.cm.service_agent_instances[service_type]
+
+    def _send_request(self, operation, request_data):
```

Comment:
All renames are suggestions to improve readability. Use as you see appropriate.
_invoke_service_agent instead of _send_request